pontoon/pretranslation/tests/test_pretranslate.py

```diff
@@ -735,38 +730,77 @@ def test_get_pretranslations_fluent_complex(
     pretranslated_string = serializer.serialize_entry(parser.parse_entry(expected))
 
     response = get_pretranslation(fluent_entity, google_translate_locale)
-    assert response == (pretranslated_string, tm_user)
+    assert response == (pretranslated_string, "tm")
+
+
+@patch("pontoon.pretranslation.pretranslate.get_google_translate_data")
+@pytest.mark.django_db
+def test_get_pretranslations_fluent_placeholders(
+    gt_mock, fluent_resource, google_translate_locale
+):
+    def gt_mock_fn(**kwargs):
+        return {"status": True, "translation": f"GT: {kwargs['text']}"}
+
+    gt_mock.side_effect = gt_mock_fn
+
+    fluent_string = dedent(
+        """
+        batman = The { $dark } Knight
+            .weapon = Brain and { -wayne-enterprise }
+            .history = Lost { 2 } parents, has { 1 } "$alfred"
+    """
+    )
+    fluent_entity = EntityFactory(resource=fluent_resource, string=fluent_string)
+
+    expected = dedent(
+        """
+        batman = GT: The { $dark } Knight
+            .weapon = GT: Brain and { -wayne-enterprise }
+            .history = GT: Lost { 2 } parents, has { 1 } "$alfred"
+    """
+    )
+
+    # Re-serialize to match whitespace
+    pretranslated_string = serializer.serialize_entry(parser.parse_entry(expected))
+
+    response = get_pretranslation(fluent_entity, google_translate_locale)
+    assert response == (pretranslated_string, "gt")
+    assert [call[1]["text"] for call in gt_mock.call_args_list] == [
+        "The {$0} Knight",
+        "Brain and {$0}",
+        'Lost {$0} parents, has {$1} "$alfred"',
+    ]
 
 
 @pytest.mark.django_db
 def test_get_pretranslations_fluent_sibling_selectors(
-    entity_a, fluent_resource, google_translate_locale, tm_user
+    entity_a, fluent_resource, google_translate_locale
 ):
     # Entity.string is a Fluent string with two sibling selectors.
     # - Move any text outside selectors into selector variants.
     # - Keep leading/trailing whitespace outside selector variants.
     TranslationMemoryFactory.create(
         entity=entity_a,
-        source="{ $key_count } key",
-        target="TM: { $key_count } key",
+        source="{ $key_count } key and { $lock_count } lock",
+        target="TM: { $key_count } key and { $lock_count } lock",
         locale=google_translate_locale,
     )
```

Comment:
The previous pretranslation behaviour did not match what's actually in the translation memory; switching to complete patterns will allow some of them to actually match.